content/tutorials/newsfeed-react.textile

```diff
@@ -0,0 +1,450 @@
+]
+
+const names = ["Ross", "Monica", "Rachel", "Joey", "Chandler", "Steve", "Bill", "Elon", "Tom", "Shaun" ]
+function getRandomArbitrary(min, max) {
+    return Math.floor(Math.random() * (max - min) + min);
+}
+```
+
+Now, let's create the Ably instance and channels. Replace '<YOUR-API-KEY>' with your own API key value. Assign the current user its name and avatar as well:
+
+```[javascript]
+    const myId = "id-" + Math.random().toString(36).substr(2, 16);
+    const apiKey = '<YOUR-API-KEY>';
+    const ably = new Ably.Realtime({
+        key: apiKey,
+        clientId: myId,
+        echoMessages: false
+    });
+    const chatChannel = ably.channels.get("chat");
+    const presenceChannel = ably.channels.get("presence");
+    let my = {};
+    my.avatar = avatarsInAssets[getRandomArbitrary(0, 9)];
+    my.name = names[getRandomArbitrary(0, 9)]
+```
+
+Now, let’s make the functional @App@ component and states within it to hold the user input as well as the updates coming in:
+
+```[javascript]
+const App = () => {
+    const [todoInput, setTodoInput] = useState("");
+    const [state, setState] = useState({
+        msgs: [],
+        newMsgs: []
+    })
+    
+    let you = {};
+}
+```
+
+Now, let’s subscribe to the channels in the @useEffect hook@ on mount of the @App@ component:
+
+```[javascript]
+useEffect(() => {
+        
+        chatChannel.subscribe("userInfo", (data) => {
+            var dataObj = JSON.parse(JSON.stringify(data));
+            if (dataObj.clientId !== myId) {
+                you.avatar = dataObj.data.avatar;
+                you.name = dataObj.data.name;
+            }
+        });
+        
+        presenceChannel.presence.subscribe('enter', function (member) {
+            if (member.clientId !== myId) {
+                chatChannel.publish("userInfo", {
+                    "avatar": my.avatar,
+                    "name": my.name
+                });
+            }
+        });
+        presenceChannel.presence.enter();
+        presenceChannel.presence.get(function (err, members) {
+            for (var i in members) {
+                if (members[i].clientId !== myId) {
+                    chatChannel.publish("userInfo", {
+                        "avatar": my.avatar,
+                        "name": my.name
+                    });
+                }
+            }
+        });
+
+        chatChannel.subscribe("chatMessage", (data) => {
+            var dataObj = JSON.parse(JSON.stringify(data));
+            var message = dataObj.data.message;
+            var date = new Date(dataObj.data.date);
+            setState(prevState => {
+                return {
+                  ...prevState,
+                  newMsgs: [...prevState.newMsgs, {"summary": message, "image": you.avatar, "date": date, "name": you.name}]
+                    };
+                });
+        })
+
+      }, []);
+```
+
+As you can see we subscribe to many channels and the publish our information. Let's discuss each of them.
```

Comment:
I see you've discussed the methods here, up to you if you want to keep them here but please add in-detail description of each, looks insufficient atm (see the previous comment).